stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/users/GroupChatInfoAddUsersViewModel.kt

```diff
@@ -0,0 +1,137 @@
+package io.getstream.chat.ui.sample.feature.chat.info.group.users
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.api.models.QueryUsersRequest
+import io.getstream.chat.android.client.call.await
+import io.getstream.chat.android.client.models.Filters
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.livedata.ChatDomain
+import kotlinx.coroutines.launch
+
+class GroupChatInfoAddUsersViewModel(
+    cid: String,
+    chatDomain: ChatDomain = ChatDomain.instance(),
+    chatClient: ChatClient = ChatClient.instance()
+) : ViewModel() {
+
+    private val channelClient = chatClient.channel(cid)
+    private var members: LiveData<List<Member>> = MutableLiveData(emptyList())
+    private val _state: MutableLiveData<State> = MutableLiveData(INITIAL_STATE)
+    private val _userAddedState: MutableLiveData<Boolean> = MutableLiveData(false)
+    private var isLoadingMore: Boolean = false
+    val state: LiveData<State> = _state
+    val userAddedState: LiveData<Boolean> = _userAddedState
+
+    init {
+        chatDomain.useCases.getChannelController(cid).enqueue { result ->
+            if (result.isSuccess) {
+                members = result.data().members
+                viewModelScope.launch {
+                    fetchUsers()
+                }
+            }
+        }
+    }
+
+    fun onAction(action: Action) {
+        when (action) {
+            is Action.UserClicked -> addMember(action.user)
+            is Action.SearchQueryChanged -> setQuery(action.query)
+            Action.LoadMoreRequested -> loadMore()
+        }
+    }
+
+    private fun addMember(user: User) {
+        viewModelScope.launch {
+            val response = channelClient.addMembers(user.id).await()
+            if (response.isSuccess) {
+                _userAddedState.value = true
+            } else {
+                // TODO: Handle error
+            }
+        }
+    }
+
+    private fun loadMore() {
+        viewModelScope.launch {
+            val currentState = _state.value!!
+
+            if (!currentState.canLoadMore || isLoadingMore) {
+                return@launch
+            }
+
+            isLoadingMore = true
+            fetchUsers()
+        }
+    }
+
+    private fun setQuery(query: String) {
+        viewModelScope.launch {
+            _state.value = State(
+                query = query,
+                results = emptyList(),
+                isLoading = true,
+                canLoadMore = true,
+            )
+            fetchUsers()
+        }
+    }
+
+    private suspend fun fetchUsers() {
+        members.value?.let { currentMembers ->
```

Comment:
Nit: `val currentMembers = members.value ?: return` would _let_ us avoid nesting the entire function body inside `let`